### ISSUE TITLE
Non-Local Usage

### ISSUE THREAD

**Issue by agahEbrahimi** [ipfs-shipyard/ipfs-pubsub-room#41]
Hello,

Does pubsub-room or the demo in the video work with two different computers on different private networks or does it just work on a private network or a singular computer? 

Thank you

**Comment by WesleyDRobinson** [comment]
Yes, it works with two different computers on different private networks.

**Comment by PierfrancescoSoffritti** [comment]
Hey, if I understand correctly, room needs to use [libp2p-webrtc-star](https://github.com/libp2p/js-libp2p-websocket-star) in order to work. libp2p-webrtc-star uses a rendezvous server in order to let nodes talk to other nodes.
Considering this, I don't understand how can all this be decentralized. If the rendezvous server goes down, nothing works anymore, right?
Am I missing something? Or my assumption about decentralization is wrong?

**Comment by pgte** [comment]
Although this is a generic aspect of IPFS / libp2p, I think I can clarify a bit:

You can use whichever transport you configure your IPFS node with.

Currently, for the browser platform, 2 main transports exist: webrtc-star and websocket-star. Both use rendezvous servers. But nothing dictates that this node only has one address. You can bind to as many rendezvous servers as you want, and thus gaining some redundancy.

(In the node.js environment you have different transport choices, with different infra-structure requirements).

**Comment by PierfrancescoSoffritti** [comment]
Thanks for the clarification @pgte.

I would like to ask you a few questions, this probably isn't the best place to do it, but I wouldn't know where else to ask.
If you don't have time to answer: no problem. Don't want to abuse your time :)

I am quite new to IPFS, I'm still trying to understand how everything comes together.
I am mostly interested in the decentralization aspect and I don't have a clear picture of how that works, yet. Is there always one or more rendezvous servers in the middle? I have seen a demo of the distributed file system, from what I understood it didn't have rendezvous entities in the middle. If not, how do nodes find each other?

I am currently working with node, not in the browser. You mentioned that there are different transport choices there, could you point me to resources relative to that?

Is there any comprehensive resource where I can read more about how everything works?

thanks in advance!

**Comment by pgte** [comment]
In the IPFS constructor you can specify which addresses and transports you want in `options.libp2p`: https://github.com/ipfs/js-ipfs#optionslibp2p

In node it default to this libp2p config:
https://github.com/ipfs/js-ipfs/blob/master/src/core/runtime/libp2p-nodejs.js

And these addresses: https://github.com/ipfs/js-ipfs/blob/master/src/core/runtime/config-nodejs.js

The libp2p packages you can use are listed here:
https://github.com/libp2p/js-libp2p#packages

Hope this helps!

**Comment by PierfrancescoSoffritti** [comment]
Great, thanks a lot!!
